### PR TITLE
feat(adapter-d1): leverage arg type for Int64 (de)serialisation

### PR DESCRIPTION
**Pull request by jkomyno**
This PR:
- fixes https://github.com/prisma/team-orm/issues/1065
- fixes https://github.com/prisma/team-orm/issues/1050

---

This PR also adjusts the client tests required after https://github.com/prisma/prisma-engines/pull/4940 was merged.
Thus, this PR fixes the integration pipeline between `prisma-engines` and `prisma`.
This fixes https://github.com/prisma/team-orm/issues/1257.

### CODE REVIEW

File: packages/driver-adapter-utils/src/types.ts [hunk 1]
```diff
@@ -28,9 +28,12 @@ export interface ResultSet {
   lastInsertId?: string
 }
 
+export type ArgType = 'Int32' | 'Int64' | 'Float' | 'Double' | 'Boolean'
+
 export type Query = {
   sql: string
   args: Array<unknown>
+  arg_types: Array<ArgType | null>
```
Comment (SevInf) [inline packages/driver-adapter-utils/src/types.ts:36]: can we `camelCase` this?

Comment (jkomyno) [reply packages/driver-adapter-utils/src/types.ts]: Yes, done!

File: packages/adapter-d1/src/utils.ts [hunk 2]
```diff
@@ -1,20 +1,38 @@
...
 
-  if (arg === false) {
-    return 0
-  }
+// Sanitize the query arguments before sending them to the database.
+export function cleanArg(arg: unknown, argType: ArgType): unknown {
+  if (arg !== null) {
+    if (argType === 'Int64') {
+      const asInt56 = Number.parseInt(arg as string)
+      if (Number.isNaN(asInt56) || Number.MAX_SAFE_INTEGER < asInt56 || Number.MIN_SAFE_INTEGER > asInt56) {
```
Comment (SevInf) [inline packages/adapter-d1/src/utils.ts:8]:
```suggestion
      if (Number.isSafeInteger(asInt56)) {
```

**Review by SevInf — APPROVED** [review]
Ok with one suggestion. once engine is merged and integrated

**Comment by jkomyno** [comment]
The engines' part of this PR is packaged and shipped in: https://github.com/prisma/prisma/pull/24981.

**Pull request merged**